ambari-server/src/main/java/org/apache/ambari/server/controller/ServiceGroupRequest.java

```diff
@@ -17,16 +17,20 @@
  */
 package org.apache.ambari.server.controller;
 
+import java.util.HashSet;
 import java.util.Objects;
+import java.util.Set;
 
 public class ServiceGroupRequest {
 
   private String clusterName; // REF
   private String serviceGroupName; // GET/CREATE/UPDATE/DELETE
+  private String stackId; // Associated stack version info
 
-  public ServiceGroupRequest(String clusterName, String serviceGroupName) {
+  public ServiceGroupRequest(String clusterName, String serviceGroupName, String stackId) {
```

Comment:
I'm a bit confused by this. Why does a service group have a stack association? An mpack has the knowledge of which stack it belongs to, so the service group should have an mpack instance association.

Reply:
@jonathan-hurley 
Our DB schemas are all coupled to stack-id. The only association we have between stack and mpack is in the stacks table. We will keep the association to stack-id to keep things consistent as well as to support creating service groups for old stacks

Reply:
But multiple mpacks could have the same stack ID, right? How does this allow us to determine which mpack a service group is associated with?

Reply:
I think using the term "stackId" is a bit confusing here - it should be differentiated from the "stackId" foreign key value. In this case, it's actually the Mpack Name and specific MPack version (with build number), right?

Reply:
Lets just call the string as "version" like we do current for cluster. 

POST /api/v1/clusters
{
  "Clusters" : {
    "cluster_name": "cl1",
    "version" : "HDP-2.5"
  }
}

Reply:
I will use version instead of stackId on rest api layer, and internally use stackId instance to  represent stackName-stackVersion.
@jonathan-hurley you are right, the only association between stack and mpack is in stack table, so if servicegroup has stackName+stackVersion, it can identify stack entity and then can get Mapck entity.(name, version etc)